src/diffusers/models/transformers/transformer_2d.py

```diff
@@ -153,91 +150,267 @@ def __init__(
+                norm_elementwise_affine,
+                norm_eps,
+                attention_type,
+                num_layers,
+                interpolation_scale,
+                caption_channels,
+            )
 
-            self.height = sample_size
-            self.width = sample_size
+        # Others.
+        self.out_channels = in_channels if out_channels is None else out_channels
+        self.gradient_checkpointing = False
 
-            self.patch_size = patch_size
-            interpolation_scale = (
-                interpolation_scale if interpolation_scale is not None else max(self.config.sample_size // 64, 1)
-            )
-            self.pos_embed = PatchEmbed(
-                height=sample_size,
-                width=sample_size,
-                patch_size=patch_size,
-                in_channels=in_channels,
-                embed_dim=inner_dim,
-                interpolation_scale=interpolation_scale,
-            )
+    def _init_continuous_input(
+        self,
+        in_channels,
+        out_channels,
+        inner_dim,
+        num_attention_heads,
+        attention_head_dim,
+        cross_attention_dim,
+        activation_fn,
+        num_embeds_ada_norm,
+        attention_bias,
+        only_cross_attention,
+        double_self_attention,
+        upcast_attention,
+        norm_type,
+        norm_elementwise_affine,
+        norm_eps,
+        attention_type,
+        norm_num_groups,
+        use_linear_projection,
+        dropout,
+        num_layers,
+    ):
+        self.use_linear_projection = use_linear_projection
 
-        # 3. Define transformers blocks
-        self.transformer_blocks = nn.ModuleList(
-            [
-                BasicTransformerBlock(
-                    inner_dim,
-                    num_attention_heads,
-                    attention_head_dim,
-                    dropout=dropout,
-                    cross_attention_dim=cross_attention_dim,
-                    activation_fn=activation_fn,
-                    num_embeds_ada_norm=num_embeds_ada_norm,
-                    attention_bias=attention_bias,
-                    only_cross_attention=only_cross_attention,
-                    double_self_attention=double_self_attention,
-                    upcast_attention=upcast_attention,
-                    norm_type=norm_type,
-                    norm_elementwise_affine=norm_elementwise_affine,
-                    norm_eps=norm_eps,
-                    attention_type=attention_type,
-                )
-                for d in range(num_layers)
-            ]
+        self.norm = torch.nn.GroupNorm(num_groups=norm_num_groups, num_channels=in_channels, eps=1e-6, affine=True)
+        if use_linear_projection:
+            self.proj_in = torch.nn.Linear(in_channels, inner_dim)
+        else:
+            self.proj_in = torch.nn.Conv2d(in_channels, inner_dim, kernel_size=1, stride=1, padding=0)
+
+        self.transformer_blocks = self._get_transformer_blocks(
```

Comment:
Since there's only a single block type used in Transformer2D at the moment,  I don't think we need a fetcher method just yet.